docs/source/preprocessing.mdx

```diff
@@ -10,209 +10,393 @@ an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express o
+
+Audio inputs are preprocessed differently than textual inputs, but the end goal remains the same: create numerical sequences the model can understand. A [feature extractor](main_classes/feature_extractor) will help you with this. Before you begin, install 🤗 Datasets to load an audio dataset to experiment with.
+
+```bash
+pip install datasets
+```
+
+Load the keyword spotting task from the [SUPERB](https://huggingface.co/datasets/superb) benchmark (see the 🤗 [Datasets tutorial](https://huggingface.co/docs/datasets/load_hub.html) for more information):
+
+```py
+>>> from datasets import load_dataset, Audio
+
+>>> dataset = load_dataset("superb", "ks")
 ```
 
-It returns a dictionary with string keys and tensor values. We can now see what the [attention_mask](glossary#attention-mask) is all about: it points out which tokens the model should pay attention to and which ones
-it should not (because they represent padding in this case).
+Access the first element of the `audio` column to take a look at the input. Calling the `audio` column will automatically load and resample the audio file:
 
+```py
+>>> dataset["train"][0]["audio"]
+{'array': array([ 0.        ,  0.        ,  0.        , ..., -0.00592041,
+        -0.00405884, -0.00253296], dtype=float32),
+ 'path': '/root/.cache/huggingface/datasets/downloads/extracted/05734a36d88019a09725c20cc024e1c4e7982e37d7d55c0c1ca1742ea1cdd47f/_background_noise_/doing_the_dishes.wav',
+ 'sampling_rate': 16000}
+```
+
+This returns three items:
 
-Note that if your model does not have a maximum length associated to it, the command above will throw a warning. You
-can safely ignore it. You can also pass `verbose=False` to stop the tokenizer from throwing those kinds of warnings.
+* `array` is the speech signal, or the input format the model expects.
+* `path` points to the location of the audio file.
+* `sampling_rate` refers to how many data points in the speech signal are measured per second.
 
-<a id='sentence-pairs'></a>
+### Resample
 
-## Preprocessing pairs of sentences
+For this tutorial, you will use the [Wav2Vec2](https://huggingface.co/facebook/wav2vec2-base) model. As you can see from the model card, the Wav2Vec2 model is pretrained on 16kHz sampled speech audio. It is important your audio data sampling rate matches the sampling rate of the dataset used to pretrain the model. If your data's sampling rate isn't the same, then you will need to resample your audio data. For example, if your audio data has a sampling rate of 48kHz, then you will need to downsample to 16kHz.
 
-<Youtube id="0u3ioSwev3s"/>
+1. Use 🤗 Datasets `cast_column` method to downsample the sampling rate to 16kHz:
 
-Sometimes you need to feed a pair of sentences to your model. For instance, if you want to classify if two sentences in
-a pair are similar, or for question-answering models, which take a context and a question. For BERT models, the input
-is then represented like this: `[CLS] Sequence A [SEP] Sequence B [SEP]`
+```py
```

Comment:
Maybe load a new datasets sampled at 48_000 kHz to that some resampling is actually happening? Maybe: 

```python
dataset = load_dataset("common_voice", "en", split="test", streaming=True)
```

Reply:
Good idea! I chose the LJ Speech dataset since it is a little simpler to show without the streaming and `IterableDataset`.